docs/en/latest/plugins/opa.md

```diff
@@ -0,0 +1,275 @@
+
+## Attributes
+
+| Name | Type | Requirement | Default | Valid | Description |
+| -- | -- | -- | -- | -- | -- |
+| host | string | required |   |   | Open Policy Agent service host (eg. https://localhost:8181) |
+| ssl_verify | boolean | optional | true |   | Whether to verify the certificate |
+| policy | string | required |   |   | OPA policy path (It is a combination of `package` and `decision`. When you need to use advanced features such as custom response, `decision` can be omitted) |
+| timeout | integer | optional | 60000ms | [1, 60000]ms | HTTP call timeout. |
+| keepalive | boolean | optional | true |   | HTTP keepalive |
+| keepalive_timeout | integer | optional | 60000ms | [1000,...] | keepalive idle timeout |
+| keepalive_pool | integer | optional | 5 | [1,...] | Connection pool limit |
+| with_route | boolean | optional | false |   | Whether to send information about the current route. |
+| with_service | boolean | optional | false |   | Whether to send information about the current service. |
+| with_consumer | boolean | optional | false |   | Whether to send information about the current consumer. (It may contain sensitive information such as apikey, so please turn it on only if you are sure it is safe) |
+
+## Data Definition
+
+### APISIX to OPA service
+
+The `type` indicates that the request type. (e.g. `http` or `stream`)
+The `reqesut` is used when the request type is `http`, it contains the basic information of the request. (e.g. url, header)
+The `var` contains basic information about this requested connection. (e.g. IP, port, request timestamp)
+The `route`, `service`, and `consumer` will be sent only after the `opa` plugin has enabled the relevant features, and their contents are same as those stored by APISIX in etcd.
+
+```json
+{
+    "type": "http",
+    "request": {
+        "scheme": "http",
+        "path": "\/get",
+        "headers": {
+            "user-agent": "curl\/7.68.0",
+            "accept": "*\/*",
+            "host": "127.0.0.1:9080"
+        },
+        "query": {},
+        "port": 9080,
+        "method": "GET",
+        "host": "127.0.0.1"
+    },
+    "var": {
+        "timestamp": 1701234567,
+        "server_addr": "127.0.0.1",
+        "server_port": "9080",
+        "remote_port": "port",
+        "remote_addr": "ip address"
+    },
+    "route": {},
+    "service": {},
+    "consumer": {}
+}
+```
+
+### OPA service response to APISIX
+
+In the response, `result` is automatically added by OPA. The `allow` is indispensable and will indicate whether the request is allowed to be forwarded through the APISIX.
+The reason, headers, and status_code are optional and are only returned when you need to use a custom response, as you'll see in the next section with the actual use case for it.
+
+```json
+{
+    "result": {
```

Comment:
Let's use 2 spaces here?

Reply:
We have used 4 spaces in the previous document, should I change it?

Reply:
No idea, just keep them is fine